File: cypress/e2e/ultimateqa.cy.cs
describe('Ultimate QA Automation Practice', () => {
  beforeEach(() => {
    cy.visit('https://ultimateqa.com/automation');
  });

  it('Interacts with the Big page with many elements', () => {
    // Click on the link to navigate to the page
    cy.contains('Big page with many elements').click();

    // Verify the new URL
    cy.url().should('include', '/complicated-page');

    // Interact with elements on the page
    // Example: Click a button
    cy.get('button').contains('Button').click();

    // Example: Fill out a form
    cy.get('input[name="et_pb_contact_name_0"]').type('John Doe');
    cy.get('input[name="et_pb_contact_email_0"]').type('john.doe@example.com');
    cy.get('textarea[name="et_pb_contact_message_0"]').type('This is a test message.');
    cy.get('button').contains('Submit').click();

    // Add more interactions as needed
  });
});

File: cypress/e2e/ultimateqa.cy.js
describe('Ultimate QA Automation - Practical 3', () => {
    beforeEach(() => {
      // Visit the Ultimate QA Automation page
      cy.visit('https://ultimateqa.com/automation');
    });
  
    it('should load the homepage correctly', () => {
      // Verify that the URL is correct
      cy.url().should('eq', 'https://ultimateqa.com/automation');
  
      // Verify that a main heading exists
      cy.get('h1').should('contain.text', 'Automation');
    });
  
    it('should navigate to a linked section and verify its content', () => {
      // Example: Click a link or button on the page
      // Replace 'Courses' with an actual link text or a more precise selector from the site.
      cy.contains('Courses').click();
  
      // Verify that the URL includes the expected path after navigation
      cy.url().should('include', '/courses');
  
      // Confirm that a heading or element unique to the Courses page is visible
      // Adjust the selector and text as necessary.
      cy.get('h1').should('exist').and('contain.text', 'Courses');
    });
  });
  